src/payment-flow-handlers/handlers/venmo-flow-handler.ts

```diff
@@ -0,0 +1,124 @@
+import { BraintreeManagerInterface } from '../../braintree-manager/braintree-interfaces';
+import { DonorContactInfo } from '../../models/common/donor-contact-info';
+import { DonationPaymentInfo } from '../../models/donation-info/donation-payment-info';
+import { PaymentProvider } from '../../models/common/payment-provider-name';
+import {
+  VenmoRestorationStateHandlerInterface,
+  VenmoRestorationStateHandler,
+} from './venmo-restoration-state-handler';
+import { DonationFlowModalManagerInterface } from '../donation-flow-modal-manager';
+
+export interface VenmoFlowHandlerInterface {
+  startup(): Promise<void>;
+  paymentInitiated(contactInfo: DonorContactInfo, donationInfo: DonationPaymentInfo): Promise<void>;
+}
+
+export class VenmoFlowHandler implements VenmoFlowHandlerInterface {
+  private donationFlowModalManager: DonationFlowModalManagerInterface;
+
+  private braintreeManager: BraintreeManagerInterface;
+
+  private restorationStateHandler: VenmoRestorationStateHandlerInterface;
+
+  constructor(options: {
+    braintreeManager: BraintreeManagerInterface;
+    donationFlowModalManager: DonationFlowModalManagerInterface;
+    restorationStateHandler?: VenmoRestorationStateHandlerInterface;
+  }) {
+    this.braintreeManager = options.braintreeManager;
+    this.donationFlowModalManager = options.donationFlowModalManager;
+    this.restorationStateHandler =
+      options.restorationStateHandler ?? new VenmoRestorationStateHandler();
+  }
+
+  /**
+   * Check if we have any results from Venmo on startup.
+   *
+   * This happens if the app redirects to us in a new tab so we can resume the session.
+   *
+   * @returns {Promise<void>}
+   * @memberof VenmoFlowHandler
+   */
+  async startup(): Promise<void> {
+    console.debug('Venmo startup');
+    const handler = await this.braintreeManager.paymentProviders.venmoHandler.get();
+    const instance = await handler?.instance.get();
+    if (instance?.hasTokenizationResult()) {
+      console.debug('Venmo startup, has tokenization results');
+
+      // if we get redirected back from venmo in a different tab, we need to restore the data
+      // that was persisted when the payment was initiated
+      const restoredInfo = await this.restorationStateHandler.getRestorationState();
+      if (restoredInfo) {
+        this.paymentInitiated(restoredInfo.contactInfo, restoredInfo.donationInfo);
+      } else {
+        console.error('no restoration info');
+        this.donationFlowModalManager.showErrorModal({
```

Comment:
do we retry?

Reply:
I don't think there's anything we can retry. We're just pulling from local storage so if the object isn't there, there's nothing we can do. The user can retry their donation.